python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -924,6 +936,15 @@ def _getattribute(self, name):
 
     return type(name, (GPIProxyObject,), d)
 
+_metaClassHolder = None
+def _getMetaClass():
+    global _metaClassHolder
+    if _metaClassHolder is None:
+        from Ganga.GPIDev.Base.Objects import ObjectMetaclass
+        _metaClassHolder = ObjectMetaclass
+    return _metaClassHolder
```

Comment:
Tbh I'm all for moving the Proxy Construction entirely away from the Object class. This is a relatively minor task (albeit a very delicate one) which I think can be accomplished in a separate PR.

I think having a pure GangaObject without knowledge of ```_proxyClass```, ```_proxyObject``` or ```_impl``` references would be nice. I don't think this should be too difficult to remove them.
Then this would require the ```addProxy```( ```GPIProxyObjectFactory``` ) to actually construct new Proxy objects on the fly (not a task too difficult).

The problem only then becomes calling functions such as ```stripProxy```, ```getName``` and ```isType``` which should be minimised within the Proxy class anyway.